packages/react-next/src/components/Checkbox/Checkbox.scss

```diff
@@ -0,0 +1,253 @@
+@import '../../common/themeCssVariables';
+@import '../../common/highContrast';
+
+:root {
+  --checkboxSize: 20px;
+  --checkboxTransitionDuration: 200ms;
+  --checkboxTransitionTiming: cubic-bezier(0.4, 0, 0.23, 1);
+}
+
+.root {
+  position: relative;
+  display: flex;
+
+  &:not(.disabled) {
+    &:not(.checked) {
+      &:hover {
+        .checkbox {
+          border-color: var(--semanticColorsInputBorderHovered);
+
+          @include high-contrast {
+            border-color: Highlight;
+          }
+        }
+      }
+
+      &:focus {
+        .checkbox {
+          border-color: var(--semanticColorsInputBorderHovered);
+        }
+      }
+
+      &:hover {
+        .checkmark {
+          color: var(--paletteNeutralSecondary);
+          opacity: 1;
+
+          @include high-contrast {
+            border-color: Highlight;
+          }
+        }
+      }
+    }
+
+    &.checked:not(.indeterminate) {
+      &:hover,
+      &:focus {
+        .checkbox {
+          background: var(--semanticColorsInputBackgroundCheckedHovered);
+          border-color: var(--semanticColorsInputBackgroundCheckedHovered);
+        }
+      }
+
+      @include high-contrast {
+        &:hover .checkbox {
+          background: Window;
+          border-color: Highlight;
+        }
+
+        &:focus .checkbox,
+        &:focus:hover .checkbox {
+          background: Highlight;
+        }
+
+        &:focus:hover .checkmark {
+          color: Window;
+        }
+
+        &:hover .checkmark {
+          color: Highlight;
+        }
+      }
+    }
+
+    &.indeterminate {
+      &:hover {
+        .checkbox,
+        .checkbox:after {
+          border-color: var(--semanticColorsInputBackgroundCheckedHovered);
+        }
+
+        .checkmark {
+          opacity: 0;
+        }
+      }
+
+      &:focus {
+        .checkbox {
+          border-color: var(--semanticColorsInputBackgroundCheckedHovered);
+        }
+      }
+    }
+  }
+}
+
+.input {
+  position: absolute;
+  background: none;
+  opacity: 0;
+}
+
+:global(.ms-Fabric--isFocusVisible) {
```

Comment:
We need to do something better here. 

We actually have this defined as a sass function already. I will dig around for it.

Reply:
thanks, found it